src/test/regress/expected/multi_subquery_union.out

```diff
@@ -23,6 +23,23 @@ LIMIT 5;
        2 |       4
 (5 rows)
 
+-- can use different filters on partition columns
+SELECT *
+FROM (
```

Comment:
`IN` is not checked. So, this test works in the master.  (`OR` also doesn't work). Seems like a bug in the previous implementation.

I think the test that started working is enough for `UNIONs`, maybe use `UNION ALL` with a `user_id=X` filters for completeness?